### PR TITLE
Fix null result while reading column values from path specified by regular expression for spark load

### PR DESCRIPTION
**Pull request by miomiocat**
If data path is specified by regular expression like `DATA INFILE("hdfs://some/cr_returned_date_sk=*/*")` while using spark load with `COLUMNS FROM PATH AS (cr_returned_date_sk)`, its column value will be parsed as `*` in `DppUtils.parseColumnsFromPath`, then it will likely be cast to wrong final output according to its type such as `NULL` for `INT` in `SparkDpp.convertSrcDataframeToDstDataframe`.

### CODE REVIEW

File: fe/spark-dpp/src/main/java/com/starrocks/load/loadv2/dpp/SparkDpp.java [hunk 1]
```diff
@@ -702,6 +702,9 @@ public boolean validateData(Object srcValue, EtlJobConfig.EtlColumn etlColumn, C
                             // append columns from path to the tail
                             List<String> columnAttributes = new ArrayList<>();
                             columnAttributes.addAll(Arrays.asList(attributes));
+                            List<String> columnValueFromPath = DppUtils.parseColumnsFromPath(record
+                                            .getAs("_starrocks_internal_input_file_path"),
+                                    fileGroup.columnsFromPath);
```
Comment (wyb) [inline fe/spark-dpp/src/main/java/com/starrocks/load/loadv2/dpp/SparkDpp.java:707]: The performance may be bad if apply function DppUtils.parseColumnsFromPath row by row.

Comment (wyb) [reply fe/spark-dpp/src/main/java/com/starrocks/load/loadv2/dpp/SparkDpp.java:707]: https://github.com/StarRocks/starrocks/blob/main/fe/spark-dpp/src/main/java/com/starrocks/load/loadv2/dpp/SparkDpp.java#L855

Is it better to convert * to the real path when getting file statistics?

Comment (miomiocat) [reply fe/spark-dpp/src/main/java/com/starrocks/load/loadv2/dpp/SparkDpp.java]: Thanks for your advice.
fixed, PTAL.

File: fe/spark-dpp/src/main/java/com/starrocks/load/loadv2/dpp/SparkDpp.java [hunk 2]
```diff
@@ -858,24 +863,18 @@ private Object convertPartitionKey(Object srcValue, Class dstClass) throws Spark
                     }
                     fileNumberAcc.add(1);
                     fileSizeAcc.add(fileStatus.getLen());
+                    dataframe = loadDataFromPath(spark, fileGroup, fileStatus.getPath().toString(), baseIndex, baseIndex.columns);
```
Comment (wyb) [inline fe/spark-dpp/src/main/java/com/starrocks/load/loadv2/dpp/SparkDpp.java:866]: Please add some comments for the update

Comment (miomiocat) [reply fe/spark-dpp/src/main/java/com/starrocks/load/loadv2/dpp/SparkDpp.java:871]: done, PTAL

**Review by wyb — APPROVED** [review]

**Review by imay — APPROVED** [review]

**Pull request merged**